ext/standard/array.c

```diff
@@ -3455,6 +3455,32 @@ PHP_FUNCTION(array_splice)
 }
 /* }}} */
 
+/* {{{ find_bucket_at_offset(HashTable* ht, zend_long offset)
+   Finds the bucket at the given valid offset */
+static inline Bucket* find_bucket_at_offset(HashTable* ht, zend_long offset)
+{
+    zend_long pos;
+    Bucket *bucket;
+    ZEND_ASSERT(offset >= 0 && offset < ht->nNumOfElements);
```

Comment:
Should this really be `< ht->nNumOfElements`? It seems to me that this might be called with `<=` as well.